Sources/SourceKitLSP/SourceKitIndexDelegate.swift

```diff
@@ -32,47 +32,51 @@ public final class SourceKitIndexDelegate: IndexDelegate {
 
   public init() {}
 
-  public func processingAddedPending(_ count: Int) {
-    queue.sync {
-      pendingUnitCount += count
+  nonisolated public func processingAddedPending(_ count: Int) {
+    queue.async {
+      await self.addPending(count)
     }
   }
 
-  public func processingCompleted(_ count: Int) {
-    queue.sync {
-      pendingUnitCount -= count
-      if pendingUnitCount == 0 {
-        _indexChanged()
-      }
+  private func addPending(_ count: Int) {
+    pendingUnitCount += count
+  }
 
-      if pendingUnitCount < 0 {
-        assertionFailure("pendingUnitCount = \(pendingUnitCount) < 0")
-        pendingUnitCount = 0
-        _indexChanged()
-      }
+  nonisolated public func processingCompleted(_ count: Int) {
+    queue.async {
+      await self.processCompleted(count)
+    }
+  }
+
+  private func processCompleted(_ count: Int) {
+    pendingUnitCount -= count
+    if pendingUnitCount == 0 {
+      _indexChanged()
+    }
+
+    if pendingUnitCount < 0 {
+      assertionFailure("pendingUnitCount = \(pendingUnitCount) < 0")
+      pendingUnitCount = 0
+      _indexChanged()
     }
   }
 
   /// *Must be called on queue*.
   func _indexChanged() {
     for delegate in mainFilesDelegates {
-      Task {
+      queue.async {
         await delegate.mainFilesChanged()
       }
     }
   }
 
   /// Register a delegate to receive notifications when main files change.
   public func registerMainFileChanged(_ delegate: MainFilesDelegate) {
-    queue.sync {
       mainFilesDelegates.append(delegate)
```

Comment:
I think this should be indented two spaces less now. Could you run swift-format on your changes?